packages/providers/src/transaction-request/transaction-request.ts

```diff
@@ -254,34 +252,36 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
    * @param predicate - Predicate bytes.
    * @param predicateData - Predicate data bytes.
    */
-  addCoinInput(coin: Coin, predicate?: AbstractPredicate) {
+  private addCoinInput(coin: CoinResource) {
     const { assetId, owner, amount } = coin;
 
-    let witnessIndex;
-
-    if (predicate) {
-      witnessIndex = 0;
-    } else {
-      witnessIndex = this.getCoinInputWitnessIndexByOwner(owner);
-
-      // Insert a dummy witness if no witness exists
-      if (typeof witnessIndex !== 'number') {
-        witnessIndex = this.createWitness();
-      }
-    }
-
     const input: CoinTransactionRequestInput = {
       ...coin,
       type: InputType.Coin,
       owner: owner.toB256(),
       amount,
       assetId,
       txPointer: '0x00000000000000000000000000000000',
-      witnessIndex,
-      predicate: predicate?.bytes,
-      predicateData: predicate?.predicateData,
+      witnessIndex: 0,
+      predicate: undefined,
+      predicateData: undefined,
```

Comment:
I reckon we can remove this explicit undefined set

Reply:
Done